src/dialogsearch.cpp

```diff
@@ -144,7 +144,21 @@ void DialogSearch::on_button_search_clicked()
     }
     else if (ui->radio_switch->isChecked())
     {
-        //TODO implement me
+        int switchID = ui->combo_switch->currentData().toInt();
+
+        search_predicate = [switchID](const RPG::EventCommand& com)
+        {
+            switch(com.code)
+            {
+                case Cmd::ControlSwitches:
+                    return (com.parameters[0] != 2 && com.parameters[1] == switchID) ||
+                        (com.parameters[0] == 1 && com.parameters[2] == switchID);
+                case Cmd::ConditionalBranch:
+                    return com.parameters[0] == 0 && com.parameters[1] == switchID;
+                default:
+                    return false;
```

Comment:
Is the Sourceline forced to integer? Otherwise you could write "Condition" in there maybe.
Btw, what is the output for "Map" when it is a common event? 0 I guess?

Maybe add some additional column to distinguish between Map event, Common event, Battle event and Precondition for Map, Common, Battle? Or **color code** them?